hibernate-reactive-core/src/test/java/org/hibernate/reactive/SQLSelectTest.java

```diff
@@ -0,0 +1,132 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+
+import org.hibernate.annotations.SQLSelect;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.reactive.testing.DBSelectionExtension;
+import org.hibernate.reactive.testing.SqlStatementTracker;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.vertx.junit5.VertxTestContext;
+import jakarta.persistence.ElementCollection;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.Id;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.POSTGRESQL;
+import static org.hibernate.reactive.testing.DBSelectionExtension.runOnlyFor;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class SQLSelectTest extends BaseReactiveTest {
+	@RegisterExtension // We use native queries, which may be different for other DBs
+	public DBSelectionExtension dbSelection = runOnlyFor( POSTGRESQL );
+
+	private SqlStatementTracker sqlTracker;
+	private Person thePerson;
+
+	// Query containing simple where check to distinguish from a possible generated query
+	private static String selectQuery = "SELECT id, name FROM person WHERE id = $1 and 'hreact' = 'hreact'";
```

Comment:
If you make this  a public constant, you can re-use it with `@SQLSelect`

Reply:
removed